### DISCUSSION TITLE
How to retrieve the username and password?

### DISCUSSION THREAD

**Discussion by Nithur-M** [plausible/analytics#2527] [Self-Hosted Support]
I've forgot the email and password for my plausible account (self hosted). I amy haave used my main email but no password change email as well.
I remember setting it when configuring the self host instance. How to retrieve it and see my email and password?
Thanks.

**Comment by ruslandoga** [comment]
@Nithur-M 👋

If you have access to the running container, you should be able to retrieve the username and email address of your user.

**Reply by Nithur-M** [reply]
yes, I have access to the container. How can I access the file that contains the username and password? thanks.

**Reply by ruslandoga** [reply]
If you are on `v1.4` you can try reading admin env vars that are set with `plausible-conf.env`:

```console
$ docker compose exec -ti plausible ash
$ echo ADMIN_USER
$ echo ADMIN_PASSWORD
```

You can also just search for `plausible.conf.env` file on your system.

---

If you are on `v1.5` you can try listing the users from the database, that won't give you the passwords, but at least you'd know the emails.

```console
$ docker compose exec -ti plausible bin/plausible remote
```
```elixir
iex> Plausible.Repo.all Plausible.Auth.User
[%Plausible.Auth.User{email: "example@email.com", ...other fields...}]
```

**Reply by Nithur-M** [reply]
Thanks for the help. I found the `env` file in `hosting/plausible.conf.env` directory.
